ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs

```diff
@@ -289,70 +308,48 @@ private static MeasureOutput MeasureText(CSSNode node, float width, float height
+
+            public static Inline Apply(CSSNode node)
             {
-                inline = new Run { Text = textNode._text };
-            }
-            else
+                return s_instance.Visit(node);
+            } 
+
+            protected sealed override Inline Make(CSSNode node, IList<Inline> children)
             {
-                var span = new Span();
-                for (var i = 0; i < length; ++i)
+                var textNode = (ReactTextShadowNode)node;
+                if (textNode._isVirtual)
+                {
+                    textNode.MarkUpdateSeen();
+                }
+
+                var text = textNode.Text;
+                if (text != null && children.Count > 0)
                 {
-                    var child = textNode.GetChildAt(i);
-                    var textChild = child as ReactTextShadowNode;
-                    if (textChild == null)
+                    throw new InvalidOperationException("Only leaf nodes can contain text.");
+                }
+                else if (text != null)
+                {
+                    var inline = new Run();
+                    inline.Text = text;
+                    FormatInline(textNode, inline, false);
+                    return inline;
+                }
+                else
+                {
+                    var inline = new Span();
+                    foreach (var child in children)
                     {
-                        throw new InvalidOperationException(
-                            $"Unexpected view type '{child.GetType()}' nested under text node.");
+                        inline.Inlines.Add(child);
                     }
 
-                    var childInline = BuildInlineFromTextCSSNode(textChild);
-                    span.Inlines.Add(childInline);
+                    FormatInline(textNode, inline, false);
+                    return inline;
                 }
-
-                inline = span;
             }
-
-            if (textNode._isColorSet)
-            {
-                inline.Foreground = new SolidColorBrush(ColorHelpers.Parse(textNode._color));
-            }
-
-            if (textNode._fontSize != UNSET)
-            {
-                var fontSize = textNode._fontSize;
-                inline.FontSize = fontSize;
-            }
-
-            if (textNode._fontStyle.HasValue)
-            {
-                var fontStyle = textNode._fontStyle.Value;
-                inline.FontStyle = fontStyle;
-            }
-
-            if (textNode._fontWeight.HasValue)
-            {
-                var fontWeight = textNode._fontWeight.Value;
-                inline.FontWeight = fontWeight;
-            }
-
-            if (textNode._fontFamily != null)
-            {
-                var fontFamily = new FontFamily(textNode._fontFamily);
-                inline.FontFamily = fontFamily;
-            }
-
-            return inline;
         }
     }
 }
```

Comment:
Got it!

---
In reply to: [56670447](https://github.com/CatalystCode/react-native/pull/273#discussion_r56670447) [](ancestors = 56670447)